retry/retry.go

```diff
@@ -0,0 +1,31 @@
+package retry
+
+import (
+	"time"
+)
+
+type Options struct {
+	Timeout    time.Duration
```

Comment:
lets call it retry interval, timeout is more like overall timeout ( for all tries all-together )

also it could accept an option of logging lambda for errors in between tries ( at least have warnings )
and probably overall timeout, or ctx